Data/Vector.hs

```diff
@@ -1776,8 +1776,19 @@ fromListN = G.fromListN
 
 -- | /O(1)/ Convert an array to a vector.
 fromArray :: Array a -> Vector a
+{-# INLINE fromArray #-}
 fromArray x = Vector 0 (sizeofArray x) x
 
+-- | /O(n)/ Convert a vector to an array.
+toArray :: Vector a -> Array a
+{-# INLINE toArray #-}
+toArray vec@(Vector offset size arr)
+  | offset == 0 && size == sizeofArray arr = arr
+  | otherwise = runST $ do
+      mvec <- unsafeThaw vec
```

Comment:
`unsafeThaw` is not needed. Using `Data.Primitive.Array.cloneArray` (backed by the `cloneArray#` primop) performs a little better because it only sets the closure type once (and on GHC 8.10 and up, avoids a write barrier when the concurrent collector is on).

Reply:
Good point, I wanted to use `cloneArray`, but saw that it wasn't available in `primitive-0.5.0.1` (current min version), but it is definitely better to just bump up the version in order to get the proper implementation.